### PR TITLE
customForcePull:  update docs for adding forcePull to custom build st…

### PR DESCRIPTION
**Pull request by gabemontero**
…rategy; also add section on custom options tha was previously absent

@bparees hey, here is the PR for the docs portion of https://trello.com/c/97Baaac6/606-2-support-force-pull-of-custom-builder-images-build

### CODE REVIEW

File: dev_guide/builds.adoc [hunk 1]
```diff
@@ -288,6 +288,67 @@ to be pulled from the registry to which the image stream points. Setting
...
+The following options are specific to the
+link:../architecture/core_concepts/builds_and_image_streams.html#custom-build[Custom
+build strategy].
+
+[[expose-docker-socket]]
+
+=== Expose Docker Socket
+
+In order to allow the running of Docker commands and the
+building of Docker images from inside the Docker container, it must be bound to an
```
Comment (bparees) [inline dev_guide/builds.adoc]: s/it/the build container/

File: dev_guide/builds.adoc [hunk 2]
```diff
@@ -288,6 +288,67 @@ to be pulled from the registry to which the image stream points. Setting
...
+  }
+}
+----
+====
+
+[[custom-force-pull]]
+
+=== Force Pull
+
+By default, when setting up the build pod, the build controller checks if  the image specified in the build configuration is
```
Comment (bparees) [inline dev_guide/builds.adoc]: extra space after "if"

File: dev_guide/builds.adoc [hunk 3]
```diff
@@ -288,6 +288,67 @@ to be pulled from the registry to which the image stream points. Setting
...
+available locally on the node.  If so,  that image will be used.  However, to override the
```
Comment (bparees) [inline dev_guide/builds.adoc]: extra space after "so,"

**Comment by bparees** [comment]
couple nits and lgtm.

@adellape this can merge once https://github.com/openshift/origin/pull/3948 merges.

**Comment by bparees** [comment]
oh and it'll need OSE exclusions until 3.0.2 ships...this feature is not in 3.0.1

File: dev_guide/builds.adoc (hunk 1 above)
Comment (gabemontero) [inline dev_guide/builds.adoc]: in next push

File: dev_guide/builds.adoc (hunk 2 above)
Comment (gabemontero) [inline dev_guide/builds.adoc]: in next push

File: dev_guide/builds.adoc (hunk 3 above)
Comment (gabemontero) [inline dev_guide/builds.adoc]: in next push

**Comment by gabemontero** [comment]
new push based on edits from Ben, including OSE exclusions via ifdef::openshift-origin[], etc.

File: dev_guide/builds.adoc [hunk 4]
```diff
@@ -288,6 +288,69 @@ to be pulled from the registry to which the image stream points. Setting
...
+  }
+}
+----
+
+<1> This flag causes the local builder image to be ignored, and a fresh version
+to be pulled from the registry to which the image stream points. Setting
+`*forcePull*` to *false* results in the default behavior of honoring the image
+stored locally.
+====
+endif::openshift-origin[]
```
Comment (bparees) [inline dev_guide/builds.adoc]: expose docker socket is in 3.0.1, so that shouldn't be if-def'd out.

Comment (gabemontero) [inline dev_guide/builds.adoc]: yep ... still include new section, just ifdef force pull

**Comment by gabemontero** [comment]
@bparees just the force pull on custom is ifdef'ed out now.

**Comment by bparees** [comment]
@adellape lgtm.  PR is on the merge queue though it looks like it hit a test failure.

**Comment by adellape** [comment]
@gabemontero @bparees Cool, LGTM. Thanks for OSE 3.0.2 exclusion, will make a note to make sure it get backs in appropriately.

**Pull request merged**